packages/pieces/airtable/src/lib/common/index.ts

```diff
@@ -14,254 +22,492 @@ export const airtableCommon = {
-              ),
-              required: false
-            }
-            if( isNil(AirtableFieldMapping[field.type]) ){
+              description: ['date', 'dateTime'].includes(field.type)
+                ? `${
+                    field.description ? field.description : ''
+                  }Expected format: mmmm d,yyyy`
+                : field.description,
+              required: false,
+            };
+            if (isNil(AirtableFieldMapping[field.type])) {
               fields[field.id] = Property.ShortText({
                 ...params,
-              })
-            }  
-            if (field.type === "singleSelect" || field.type === "multipleSelects") {
-              const options = field.options?.choices.map((option: { id: string, name: string }) => ({
-                value: option.id,
-                label: option.name
-              }))
-  
-              fields[field.id] = (AirtableFieldMapping[field.type])({
+              });
+            }
+            if (
+              field.type === 'singleSelect' ||
+              field.type === 'multipleSelects'
+            ) {
+              const options = field.options?.choices.map(
+                (option: { id: string; name: string }) => ({
+                  value: option.id,
+                  label: option.name,
+                })
+              );
+
+              fields[field.id] = AirtableFieldMapping[field.type]({
                 ...params,
                 options: {
-                  options: options ?? []
-                }
-              })
+                  options: options ?? [],
+                },
+              });
             } else {
-              fields[field.id] = (AirtableFieldMapping[field.type])(params)
+              fields[field.id] = AirtableFieldMapping[field.type](params);
             }
-          } 
-        })
+          }
+        });
       } catch (e) {
-        console.debug(e)
+        console.debug(e);
       }
 
-      return fields
-    }
+      return fields;
+    },
   }),
-  
-  async createNewFields( auth : string , base : string , tableId : string , fields : Record<string,unknown>){
-    if( !auth ) return fields;
-    if( !base ) return fields;
-    if( !tableId ) return fields;
 
- 
-    const newFields : Record<string,unknown> = {};
+  fieldNames: Property.DynamicProperties({
+    displayName: 'Table',
+    required: true,
+    refreshers: ['base', 'tableId'],
+
+    props: async ({ auth, base, tableId }) => {
+      if (!auth) return {};
+      if (!base) return {};
+      if (!tableId) return {};
+
+      let fieldNames = {};
+
+      try {
+        const airtable: AirtableTable = await airtableCommon.fetchTable({
+          token: auth as unknown as string,
+          baseId: base as unknown as string,
+          tableId: tableId as unknown as string,
+        });
 
-    try{
+        fieldNames = airtable.fields.map((field: AirtableField) => field.name);
+      } catch (e) {
+        console.debug(e);
```

Comment:
I think it's better to make it throw the error, so the user can know in case there is problem instead of blank field names

What do you think?

Reply:
Also the same thing in above files

Reply:
bumped + removed try/catch